website_lazy_load_image/tests/test_lazy_load_image.py

```diff
@@ -93,8 +97,20 @@ def test_encoding_render(self):
         ui_view = self.env['ir.ui.view'].sudo(
             public_user_id).with_context(website_id=self.website_id)
         res = ui_view.render_template(self.view_4.id).decode('UTF-8')
-        arch = '<main><span>Teléfono, means phone</span></main>'
+        arch = '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN" ' \
+               '"http://www.w3.org/TR/REC-html40/loose.dtd">\n' \
+               '<main><span>Teléfono, means phone</span></main>'
         self.assertEqual(res, arch)
         robots = self.env.ref('website.robots').render()
-        self.assertNotIn('<html>', robots.decode(
-            'UTF-8'), "Robots must not be wrapped into html DOM")
+        self.assertNotIn('<html>', robots.decode('UTF-8'),
+                         "Robots must not be wrapped into html DOM")
+
+    def test_doctype_in_main(self):
+        """ Check that at least doctype is preserved on website """
+        public_user_id = self.ref('base.public_user')
+#       TODO: Improve this part and find a way to render website template
```

Comment:
what's the problem? If it's the request not bound, you can use https://github.com/odoo/odoo/blob/master/addons/website/tools.py#L76

Reply:
@simahawk , as it's my first test ever on Odoo I was not aware of this mockup class. I'll use it later on.